### PR TITLE
sandbox: Fail to start if a time mode is not explicitly specified.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
I suggest hiding whitespace changes when reviewing.

### Changelog

- **[Sandbox]** Sandbox is switching from Static Time mode to Wall Clock Time mode as the default. To ensure that our users know about this, for one version, there will be no default time mode. Instead, users will have to explicitly select their preferred time mode by means of the `--static-time` or `--wall-clock-time` switches. In the next release, Wall Clock Time will become the default, and users who are happy with the defaults will no longer need to specify the time mode.
- **[DAML Assistant]** Initializing a new DAML project adds a switch to `daml.yaml` to ensure Sandbox can continue to start with `daml start`:

  ```
  sandbox-options:
    - --wall-clock-time
  ```

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
@moritzkiefer-da: Thank you, I was going to ask today what needs to be done there but you're way ahead of me. 😄

File: docs/source/tools/sandbox.rst [hunk 1]
```diff
@@ -10,13 +10,15 @@ The DAML Sandbox, or Sandbox for short, is a simple ledger implementation that e
...
+It is possible to execute the Sandbox launching step in isolation by typing ``daml sandbox --wall-clock-time``.
+
+Note: Sandbox is currently in the process of moving from Static Time mode to Wall Clock Time mode as the default. For this version only, you need to specify the time mode explicitly when launching Sandbox outside ``daml start``. (When you use ``daml start``, it provides this flag and any other flags in the ``sandbox-options:`` section of ``daml.yaml``.) From the next release, you can omit this flag.
 
 Sandbox can also be run manually as in this example:
 
 .. code-block:: none
 
-  $ daml sandbox Main.dar --scenario Main:example
+  $ daml sandbox Main.dar --static-time --scenario Main:example
```
Comment (cocreature) [inline docs/source/tools/sandbox.rst]: Any reason why we want static time here?

File: docs/source/triggers/template-root/daml.yaml.template [hunk 2]
```diff
@@ -11,3 +11,5 @@ dependencies:
   - daml-stdlib
   - daml-trigger
 # trigger-dependencies-end
+sandbox-options:
```
Comment (cocreature) [inline docs/source/triggers/template-root/daml.yaml.template]: The docs for triggers and DAML script currently tell you to pass `--static-time` to `daml script` and `daml trigger` (I switched them over to requiring the time to be specified explicitly a while back already). With this change that is inconsistent. There is no reason for why they need static-time, I simply didn’t want to deviate from the default so the best fix is to use --wall-clock-time in the docs as well.

Comment (SamirTalwar-DA) [reply docs/source/triggers/template-root/daml.yaml.template]: OK, but even if you're using `daml start`? If you use `daml sandbox`, you need to use `--wall-clock-time`.

File: docs/source/tools/sandbox.rst (hunk 1 above)
Comment (SamirTalwar-DA) [reply docs/source/tools/sandbox.rst]: Scenarios can't be run under wall clock time. (TBH we should probably just remove this section and tell people to use DAML Script.) It's explained on line 30.

**Comment by bame-da** [comment]
My preference would be to make this configurable through `daml.yaml` have it set to wall clock in all our examples. and have both `daml start` and `daml sandbox` pick that up.

File: docs/source/triggers/template-root/daml.yaml.template (hunk 2 above)
Comment (cocreature) [reply docs/source/triggers/template-root/daml.yaml.template]: This is about the argument passed to the script runner and the trigger runner. With this change running `daml start` and the starting the trigger runner with `daml trigger --static-time` is definitely not going to work. So all that’s necessary is to also flip the mode in the invocations to `daml trigger` and `daml script`.

**Comment by cocreature** [comment]
> My preference would be to make this configurable through `daml.yaml` have it set to wall clock in all our examples. and have both `daml start` and `daml sandbox` pick that up.

So far `daml sandbox` does not pick up anything. So it also doesn’t pick up DARs which `daml start` will do. I’m a bit hesitant to add magic in the assistant around `daml sandbox`. IMHO it makes more sense for it to just call sandbox directly.

File: docs/source/tools/sandbox.rst (hunk 1 above)
Comment (cocreature) [reply docs/source/tools/sandbox.rst]: Oh of course, I completely missed the fact that this uses a scenario.

File: docs/source/triggers/template-root/daml.yaml.template (hunk 2 above)
Comment (SamirTalwar-DA) [reply docs/source/triggers/template-root/daml.yaml.template]: Done.

**Comment by SamirTalwar-DA** [comment]
Hmmmm… I'd like to rejig the quickstart guide to use Wall Clock Time mode and DAML Script rather than scenarios, but it makes it more complicated because we can't just say "the contract ID is #2:1", because heartbeats.

Soon we'll get rid of heartbeats, I guess, so perhaps this can wait and I'll tell people to use `--static-time` explicitly for now. But that sucks, because the quickstart should be representative of good practices. Or I could make it clear that the numbers might vary and update the guide now.

Thoughts, anyone?

**Comment by cocreature** [comment]
The quickstart guide already uses DAML script (otherwise we wouldn’t have deprecated scenarios).

I think I’d rather explain contract ids varying than use static time. Especially given that we’re making contract ids much more complicated the current scheme isn’t going to work anyway.

@bame-da Any opinions on this?

**Comment by SamirTalwar-DA** [comment]
Missed that. Serves me right for using docs.daml.com rather than building them myself.

**Comment by SamirTalwar-DA** [comment]
Updated the quickstart (and the manual release script) and added a paragraph on contract IDs. Let me know what you think!

File: docs/source/tools/sandbox.rst (hunk 1 above)
Comment (gerolf-da) [reply docs/source/tools/sandbox.rst]: I'd say let's remove everything about scenarios and add a link to DAML Triggers for initializing the ledger. Sorry to have missed this bit of text when deprecating scenarios the last time around.

**Pull request merged**